app/services/discussion_service.rb

```diff
@@ -40,22 +40,14 @@ def self.update(discussion:, params:, actor:)
     end
 
     return false unless discussion.valid?
-
-    update_search_vector = discussion.title_changed? || discussion.description_changed?
-
-    event = true
-    if discussion.title_changed?
-      event = Events::DiscussionTitleEdited.publish!(discussion, actor)
-    end
-
-    if discussion.description_changed?
-      event = Events::DiscussionDescriptionEdited.publish!(discussion, actor)
-    end
-
+    return discussion if discussion.changed == ['uses_markdown']
```

Comment:
Yea I've been holding rules in my head but it's great to discuss them for real.

Service methods return:
true, record or event if all is well.
false if record validation fails.
and they raise an exception if you do no have permission to perform the action.